docs/manual/developer/04_style_guide.md

```diff
@@ -0,0 +1,229 @@
+    * `dependency`
+    * `performance`
+    * `management`
+    * `functionality`
+* `conflicts`
+  * Must be a valid rule id
+* `requires`
+  * * Must be a valid rule id
+* `template`
+
+### Group
+This section describes the styleguide around the `group.yml` files. 
+All the above [YAML](manual/developer/04_style_guide:yaml) rules apply.
+A group should only contain rules that effect the same software or service.
+
+#### Group Sections
+Group sections must be in the following order, if present.
+* `documentation_complete`
+* `title`
+  * Must be in [Title case](https://en.wikipedia.org/wiki/Title_case)
+* `platforms`
+* `description` (HTML-Like)
+
+### Benchmark
+This section describes the styleguide around the `benchmark.yml` files. 
+All the above [YAML](manual/developer/04_style_guide:yaml) rules apply.
+
+#### Benchmark Sections
+Benchmark sections must be in the following order, if present.
+* `documentation_complete`
+* `status`
+* `title`
+  * Must be in [Title case](https://en.wikipedia.org/wiki/Title_case)
+* `description` (HTML-Like)
+* `notice` (HTML-Like)
+* `front-matter` (HTML-Like)
+* `rear-matter` (HTML-Like)
+* `version`
+
+### Controls
+These rules apply to the files in `*.yml`
+All the above [YAML](manual/developer/04_style_guide:yaml) rules apply.
+
+### Control Sections
+Control sections must be in the following order, if present.
+* `policy`
+* `title`
+  * Must be in [Title case](https://en.wikipedia.org/wiki/Title_case)
+* `id`
+  * Must be short
+  * Must be lowercase
+  * If product specific should be in the format `standard_product`. For example CIS on RHEL8 would be `cis_rhel8`
+  * Should match the filename of the control
+  * Words shall be separated by an underscore
+* `version`
+  * Should be the same as the standard
+* `source`
+  * URL to the standard
+* `controls`
+  * `id`
+  * `levels`
+    * Should be in lowercase
+    * Must have words separated by an underscore
+    * Shall follow the standard
+  * `title`
+    * Shall be one line
+  * `status`
+  * `notes`
+    * Must be a block
+  * `rules`
+    * Must be a valid rule id
+
+## Remediation
+All remediations should have the following header with the appropriate values 
+```bash
+# platform = multi_platform_all
+# reboot = false
+# strategy = enable
+# complexity = low
+# disruption = low
+```
+
+### Bash
+* Should use macros instead shared functions
+* Shall use 4-space indentation
+* Put `do` or `then` on the same line as `for` or `if` respectively, e.g. `for file in *; do`
+
+### Ansible
+* Shall follow all the rules in the [YAML](manual/developer/04_style_guide:yaml) section
+* Should prefer using Ansible modules over just calling system commands
+* Shall be written to pass `ansible-lint`
+
+### Kubernetes
+* Shall follow all the rules in the [YAML](manual/developer/04_style_guide:yaml) section
+
+## XML
+* Shall use the `.xml` for the file extension
+* Must be able to be parsed by Python's XML parser
+* Shall use 4-space indentation
+
+### OVAL
```

Comment:
Do we want to add something about OVAL comments here? IMO they're really helpful and while mandated in certain places, aren't mandated everywhere and thus we should encourage their use and clarity?